src/XrdEc/XrdEcReader.cc

```diff
@@ -799,8 +485,13 @@ namespace XrdEc
       uint32_t crc32val = objcfg.digest( 0, buffer, lfh.uncompressedSize );
       if( crc32val != lfh.ZCRC32 ) return false;
       // keep the metadata
-      std::string url = objcfg.GetDataUrl( std::stoull( lfh.filename ) );
-      metadata.emplace( url, buffer_t( buffer, buffer + lfh.uncompressedSize ) );
+      try{
+    	  std::string url = objcfg.GetDataUrl( std::stoull( lfh.filename ) );
+    	  metadata.emplace( url, buffer_t( buffer, buffer + lfh.uncompressedSize ) );
+      }
+      catch(const std::invalid_argument* e){
```

Comment:
I suppose this is to handle the `stoull` but I'm rather sure it wont throw the exception by pointer, also shouldn't we fail the `ParseMetada` in this case? or at least log the error?